src/Broadway/Snapshotting/SnapshotStore.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * This file is part of the broadway/broadway package.
+ *
+ *  (c) Qandidate.com <opensource@qandidate.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace Broadway\Snapshotting;
+
+interface SnapshotStore
+{
+    /**
+     * @param Snapshot $snapshot
+     */
+    public function save(Snapshot $snapshot);
+
+    /**
+     * @param mixed $id
+     *
+     * @return Snapshot
+     *
+     * @throws SnapshotNotFoundException
```

Comment:
I considered this as well but for whatever reason I didn't go through with it in the end. I do agree that it makes things a lot cleaner.